### PR TITLE
Add `refreshIntervalTemplate` to `OpenDataSoftCatalogItemTraits`

### LINKED ISSUE #6607: Add polling/auto refreshing support for Opendatasoft
Some ODS portals publish data that gets updated at regular frequency - for example - live temperature data.

For example - https://data.ballarat.vic.gov.au/explore/dataset/noise-observations/information/ has a metadata field called "Update frequency".

This metadata should be used to set `refreshInterval`

Eg

image

### PR DESCRIPTION
**Pull request by nf-s**
### Add `refreshIntervalTemplate` to `OpenDataSoftCatalogItemTraits`

Fixes https://github.com/TerriaJS/terriajs/issues/6607

Add `refreshIntervalTemplate` to `OpenDataSoftCatalogItemTraits` - this can be used to set `refreshInterval` using Mustache template rendered on ODS Dataset JSON object

**Note** this approach could be used for other catalog items.

### Test me

For example - https://data.ballarat.vic.gov.au/explore/dataset/noise-observations/information/ has a metadata field called "Update frequency".

This appears in the `dataset` object at path `metas.custom.update-frequency` https://data.ballarat.vic.gov.au/api/v2/catalog/datasets/air-quality-observations/

Example config - https://gist.githubusercontent.com/nf-s/f1af76253c9a06130e9d24dc999d3c37/raw/b0d38fd8ac0ef69617d96ccaed3ca6190ee452c3/ods-refreshIntervalTemplate.json

```json
{
  "catalog": [
    {
      "name": "Ballarat",
      "url": "https://data.ballarat.vic.gov.au",
      "id": "2e99d97b-a9ae-4476-bfa2-91db69408920",
      "type": "opendatasoft-group",
      "itemProperties": {
        "refreshIntervalTemplate": "{{metas.custom.update-frequency}}"
      },
      "facetFilters": [
        {
          "name": "features",
          "value": "geo"
        }
      ]
    }
  ]
}
```

#### Test link

http://ci.terria.io/ods-group-refresh/#clean&https://gist.githubusercontent.com/nf-s/f1af76253c9a06130e9d24dc999d3c37/raw/b0d38fd8ac0ef69617d96ccaed3ca6190ee452c3/ods-refreshIntervalTemplate.json

### Checklist

- [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Models/Catalog/CatalogItems/OpenDataSoftCatalogItem.ts [hunk 1]
```diff
@@ -447,6 +446,29 @@ export class OpenDataSoftDatasetStratum extends LoadableStratum(
...
+  @computed get refreshInterval() {
+    if (!this.catalogItem.refreshIntervalTemplate) return;
+
+    try {
+      const string = Mustache.render(
+        this.catalogItem.refreshIntervalTemplate,
+        this.dataset
+      );
+      if (isJsonString(string)) {
+        const timeInMs = (ms(string) || 0) / 1000;
```
Comment (mwu2018) [inline lib/Models/Catalog/CatalogItems/OpenDataSoftCatalogItem.ts]: The variable timeInMs Should be renamed as timeInSecs to improve code readability.

File: wwwroot/test/ods/datasets.json [hunk 2]
```diff
@@ -555,6 +555,9 @@
             "oauth_scope": null,
             "federated": false,
             "license_url": "https://creativecommons.org/licenses/by/4.0/"
+          },
+          "custom": {
+            "update_frequency": "10 min"
```
Comment (mwu2018) [inline wwwroot/test/ods/datasets.json]: Not being used.

**Review by mwu2018 — CHANGES_REQUESTED** [review]
Some minor changes required.

File: wwwroot/test/ods/datasets.json (hunk 2 above)
Comment (nf-s) [reply wwwroot/test/ods/datasets.json]: Done!

**Review by mwu2018 — APPROVED** [review]

**Pull request merged**